components/hab/src/scaffolding.rs

```diff
@@ -0,0 +1,167 @@
+// Copyright (c) 2016-2017 Chef Software Inc. and/or applicable contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+use std::str::FromStr;
+use std::path::Path;
+use std::io;
+use std::fs;
+
+use error::Result;
+
+use hcore::crypto::init;
+use hcore::package::PackageIdent;
+use common::ui::{UI, Status};
+
+// Check to see if the --scaffolding passed matches available core scaffolding
+// If not check if we've been given a pkg ident for a custom scaffolding
+pub fn scaffold_check(ui: &mut UI, maybe_scaffold: Option<&str>) -> Result<Option<PackageIdent>> {
+    match maybe_scaffold {
+        Some(scaffold) => {
+            init();
+            match scaffold {
```

Comment:
I would downcase this before attempting to match so you don't have a case sensitive match

```
scaffold.to_lowercase()
```